### ISSUE TITLE
SickRage samples

### ISSUE THREAD

**Issue by Toilal** [guessit-io/guessit#222]
All SickRage samples should work in guessit (https://github.com/SiCKRAGETV/SickRage/blob/master/sickbeard/name_parser/regexes.py)

Currently failing : 
- [x] ~~`tpz-abc102`: pattern is named as stupid in sickrage, it works in guessit with `tpz-abc-102` with option `-t episode`.~~
- [x] `Show.Name.Part.3.HDTV.XViD.Etc-Group`: use `-t episode` option, because we can't assume it's a series (some movies have Parts)
- [x] `Show.Name.Part.1.and.Part.2.Blah-Group`: part property should have the same logic as episode number to build a `partList` property.
- [x] `[ISLAND]One_Piece_679_[VOSTFR]_[V1]_[8bit]_[720p]_[EB7838FC].mp4`: Must use -E option 

```
-E, --episode-prefer-number
                        Guess "serie.213.avi" as the episodeNumber 213.
                        Without this option, it will be guessed as season 2,
                        episodeNumber 13
```
- [x] `[ISLAND]One_Piece_679_[VOSTFR]_[V1]_[8bit]_[720p]_[EB7838FC].mp4`: V1 is guessed as a title.
- [x] `[Kaerizaki-Fansub]_One_Piece_679_[VOSTFR][HD_1280x720].mp4`: Because `Kaerizaki-Fansub` contains `Fansub`, it is not guessed as the release group and `Kaerizaki` is guessed as the series name, instead of `One Piece`.
- [x] `[Group Name] Show Name.13-14`: Only the last `episodeNumber` (14) is guessed as an episode number, `13` belongs to the series name.
- [x] `[ShinBunBu-Subs] Bleach - 02-03 (CX 1280x720 x264 AAC)`: Doesn't guess an `episodeList`. `03` is guessed as the title.
- [ ] `Bleach - s16e03-04 - 313-314`: 313-314 is guessed as title, absolute episodes are not supported (related to #142)
- [x] `Bleach s16e03e04 313-314`: Fails to grab proper `episodeNumber` and `episodeList` 

```
? Bleach s16e03e04 313-314
: episodeList:
  - 4
  - 313
  - 314
  episodeNumber: 3
  season: 16
  series: Bleach
  type: episode
```
- [ ] `165.Naruto Shippuuden.s08e014`: Only grabs relative episodeNumber `14`, 165 is attached to series title.
- [x] `003-004. Show Name - Ep Name.ext`: Doesn't guess an `episodeList`.

**Comment by OmgImAlexis** [comment]
@miigotu @webspider Keep an eye on this.

**Comment by WebSpider** [comment]
Subscribed.

**Comment by Toilal** [comment]
Some tips for SickRage developers :

As SickRage is an episode tool, it should run guessit using `-t episode` option. (Equivalent to use shortcut API method [guess_episode_info](https://github.com/wackou/guessit/blob/master/guessit/__init__.py#L357))
```
-t TYPE, --type TYPE  The suggested file type: movie, episode. If undefined,
                        type will be guessed.
```

 Also, if series name is known at parse time, you can use `-S --expected-series` option, it can help in some cases.
```
-S EXPECTED_SERIES, --expected-series EXPECTED_SERIES
                        Expected series to parse (can be used multiple times)
```

Also, `-E --episode-prefer-number` option can be useful. This option should be configurable by the user : on some sources, it's better to let it disabled, and in other cases it must be enabled. Maybe others options should be configurable too, but this one seems mandatory to me.

```
-E, --episode-prefer-number
                        Guess "serie.213.avi" as the episodeNumber 213.
                        Without this option, it will be guessed as season 2,
                        episodeNumber 13
```

All command line options are available through the API. Pass [options values](https://github.com/wackou/guessit/blob/master/guessit/options.py) to [API methods](https://github.com/wackou/guessit/blob/master/guessit/__init__.py#L263-L357) through options dict kwarg. Key of the dict is the option name, as declared in ArgumentParser `dest` property. 

General options are defined in [options.py](https://github.com/wackou/guessit/blob/master/guessit/options.py) files, but transformer plugins can also register options (ie. `-E --episode-prefer-number` option is registered in [transfo/guess_episodes_rexps.py](https://github.com/wackou/guessit/blob/master/guessit/transfo/guess_episodes_rexps.py#L180-L183)).

**Comment by miigotu** [comment]
We dont use guessit command line, we use it as a library.

**Comment by Toilal** [comment]
@miigotu I've explained at the end of the previous message how to pass options using the API (=library).

But translating from command line options to dict API options is not easy and undocumented, so i'll add possibility to pass options as a string and get it parsed by ArgumentParser even from the API.

**Comment by Toilal** [comment]
From this commit https://github.com/wackou/guessit/commit/05f4f9e2a616f5ec13e71a17b2e0c907fa375746, options can be passed exactly like in command line shell.

```
(3.4.3-guessit) toilal@Toilal-Ubuntu:~/IdeaProjects/guessit$ python
Python 3.4.3 (default, Jul  4 2015, 12:12:53) 
[GCC 4.8.4] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import guessit

>>> guessit.guess_file_info("[ISLAND]One_Piece_679_[VOSTFR]_[V1]_[8bit]_[720p]_[EB7838FC].mp4", options="-t episode -E")
{'type': 'episode', 'series': 'One Piece', 'screenSize': '720p', 'title': 'V1', 'releaseGroup': 'ISLAND', 'videoProfile': '8bit', 'mimetype': 'video/mp4', 'subtitleLanguage': [<Language [fr]>], 'episodeNumber': 679, 'crc32': 'EB7838FC', 'container': 'mp4'}

>>> guessit.guess_episode_info("[ISLAND]One_Piece_679_[VOSTFR]_[V1]_[8bit]_[720p]_[EB7838FC].mp4", options="-E")
{'type': 'episode', 'series': 'One Piece', 'screenSize': '720p', 'title': 'V1', 'releaseGroup': 'ISLAND', 'videoProfile': '8bit', 'mimetype': 'video/mp4', 'subtitleLanguage': [<Language [fr]>], 'episodeNumber': 679, 'crc32': 'EB7838FC', 'container': 'mp4'}

```

**Comment by Toilal** [comment]
Most of case are now working (see checked cases in the 1st post). The remaining issues to fix are about absolute episode support (#142).

**Comment by WebSpider** [comment]
Subscribing to that issue too, so we wont miss any related update ;)

-----Original Message-----

**Comment by Toilal** [comment]
New release 0.11.0 is in nature ! 

It still doesn't support relative/absolute episode numbering (it will only grab the first episodes found), and doesn't grab episode numbers from `tpz-abc102` pattern. Are those things really required for sickrage ?

**Comment by OmgImAlexis** [comment]
@miigotu this should be fine now for all but anime and sports since they're the only ones I know that need relative/absolute numbering.

**Comment by miigotu** [comment]
relative and absolute numbering works fine already @OmgImAlexis . tpz-abc102 is not a scene naming method in any genre, which is why SR calls it stupid and we really dont care if non-scene-conforming names don't work. Philosophy being that if the uploader doesnt know scene naming standards then their video likely is not scene conforming either. We need some sort of quality control and as it stands, the scene naming schemes are our all we have before downloading.

**Comment by OmgImAlexis** [comment]
`zoo.11112.hdtv-lol.mkv` doesn't match the episode numbers.
Should return season 01, episode 11, and 12.

````
For: zoo.11112.hdtv-lol.mkv
GuessIt found: {
    [1.00] "mimetype": "video/x-matroska", 
    [1.00] "releaseGroup": "lol", 
    [1.00] "container": "mkv", 
    [0.60] "title": "zoo 11112", 
    [1.00] "format": "HDTV", 
    [1.00] "type": "movie"
}
````

**Comment by miigotu** [comment]
@OmgImAlexis imho it should say FU thats not a real scene name.

**Comment by Toilal** [comment]
Yes, it could also mean season season 11, episode 112, or anything else. We can't support those kind of filenames.

**Comment by OmgImAlexis** [comment]
@miigotu unless SR specifically doesn't download these kinds of files we kind of have to support it. 

@Toilal is there no way at all this could be supported?

**Comment by miigotu** [comment]
@OmgImAlexis afaik SR doesn't download file names it cannot parse.

**Comment by OmgImAlexis** [comment]
@miigotu the name of the torrent is fine, it's just the file that isn't. So unless SR is opening .torrents it won't know that it's "wrong" until the file's already downloaded.

**Comment by miigotu** [comment]
Is the resulting file in a subfolder, with the folder having the right name?

**Comment by OmgImAlexis** [comment]
@miigotu I don't think so, just the file.

**Comment by WebSpider** [comment]
It would make sense to have guessit try and parse what we decided to download, in addition to the file on disk, after unrar or something.

Btw: How did the postprocessor ever figure out what to do with the media ... Do you have log files from that?

-----Original Message-----

**Comment by Toilal** [comment]
If name of the `.torrent` is OK, why not trying to use this name too to try to retrieve the information ?

As sickrage is an episode only tool (no movies), you know that you have to parse an episode, so you can use `-t episode` option to disable filetype guessing and force episode filetype.

@OmgImAlexis => What if `010203` ? Is it season `01`, episode `02` & `03`, or episode `01`, `02` & `03` ? And why not season`03`, episode `01` & `02` ?

**Comment by OmgImAlexis** [comment]
@Toilal in all my time using KAT and TPB I've only ever seen it in this format `11112`.
Season should be 1 or 2 digits as hardly any shows have more than 99 seasons apart from anime which should not use this naming pattern. Episodes have 2 digits so we can assume it's split like this `1` `11` `12` so it'd be season 1, anything after that should be 2 digit groups of episodes so `11` and `12`.

**Comment by OmgImAlexis** [comment]
@WebSpider it hasn't yet. Still waiting for it to download, I'll move it into my process directory soon and post the logs. From memory the old regex we use still handles this "okay".

**Comment by Toilal** [comment]
@OmgImAlexis then I think it's possible if we stick to this exact pattern. Actually, guessit works in this case when there's a single episode number **with** `-t episode` option ([because some movies have 3/4 digits inside their name](https://en.wikipedia.org/wiki/List_of_films:_numbers#100-999)), but **without** option `--episode-prefer-number`.

```
-E, --episode-prefer-number
                        Guess "serie.213.avi" as the episodeNumber 213.
                        Without this option, it will be guessed as season 2,
                        episodeNumber 13
```

**Comment by OmgImAlexis** [comment]
You mean like so? 

````
➜  ~  guessit -t episode --episode-prefer-number "zoo.11112.hdtv-lol.mkv"
For: zoo.11112.hdtv-lol.mkv
GuessIt found: {
    [1.00] "mimetype": "video/x-matroska", 
    [1.00] "releaseGroup": "lol", 
    [1.00] "container": "mkv", 
    [1.00] "format": "HDTV", 
    [0.40] "series": "zoo 11112", 
    [1.00] "type": "episode"
}
````

**Comment by miigotu** [comment]
We dont even use guessit for post processing. We only use guessit for subtitles right now.

**Comment by Toilal** [comment]
@OmgImAlexis please read carefully what i've written in my previous post (`with`!=`without`)

Currently guessit works with those cases and options
```
guessit -t episode  zoo.112.hdtv-lol.mkv
For: zoo.112.hdtv-lol.mkv
GuessIt found: {
    [0.42] "series": "zoo",
    [1.00] "container": "mkv",
    [0.36] "episodeNumber": 12,
    [1.00] "type": "episode",
    [1.00] "mimetype": "video/x-matroska",
    [0.60] "releaseGroup": "lol",
    [0.60] "format": "HDTV",
    [0.36] "season": 1
}

guessit -t episode --episode-prefer-number zoo.112.hdtv-lol.mkv
For: zoo.112.hdtv-lol.mkv
GuessIt found: {
    [0.60] "releaseGroup": "lol",
    [0.24] "episodeNumber": 112,
    [0.42] "series": "zoo",
    [1.00] "container": "mkv",
    [1.00] "type": "episode",
    [0.60] "format": "HDTV",
    [1.00] "mimetype": "video/x-matroska"
}
```

And maybe we could support the next one like this =>

```
guessit -t episode zoo.11213.hdtv-lol.mkv
For: zoo.11213.hdtv-lol.mkv
GuessIt found: {
    [0.42] "series": "zoo",
    [1.00] "container": "mkv",
    [0.36] "episodeNumber": 12,
    [0.36] "episodeList": [12, 13]
    [1.00] "type": "episode",
    [1.00] "mimetype": "video/x-matroska",
    [0.60] "releaseGroup": "lol",
    [0.60] "format": "HDTV",
    [0.36] "season": 1
}
```

**Comment by ratoaq2** [comment]
Last pending scenario fixed by #491
